pnt-lut.c

```diff
@@ -62,21 +66,44 @@ int main(int ac, char *av[]) {
     mkfifo(pnt_fifo, 0666);
     const int fd = open(pnt_fifo, O_RDONLY);
     const int fd_out = open(output_fifo, O_WRONLY);
-    uint8_t pnt[2];
+    int pnt[2];
     for (;;) {
       const int rv = read(fd, pnt, sizeof(pnt));
-      if (rv != sizeof(pnt))
+      if (rv != sizeof(pnt)) {
+        if (rv != 0) {
+          syslog(LOG_INFO, "point read failed, rv=%d\n", rv);
+        }
         continue;
+      }
+
+      int tablet_x = pnt[0], tablet_y = pnt[1];
+
+      float pgm_x = PGM_W - tablet_y*PGM_W/TABLET_MAX_Y;
+
+      float pgm_y = tablet_x*PGM_H/TABLET_MAX_X;
 
-      uint8_t x = pnt[0], y = pnt[1];
+#ifdef DEBUG
+      syslog(LOG_DEBUG, "tablet x is %d", tablet_x);
+      syslog(LOG_DEBUG, "tablet y is %d", tablet_y);
+      syslog(LOG_DEBUG, "calculated pgm_x is %f", pgm_x);
+      syslog(LOG_DEBUG, "calculated pgm_y is %f", pgm_y);
+#endif
 
       // output to haptic driver pipe
       uint8_t output[2];
       output[0] = 0; // real-time playback mode
       // offset 15
       // + row size * number of rows (y)
       // + distance into row (x)
-      output[1] = global[15 + y * PGM_W + x]; // 0-255 strength of vibration
+      if (tablet_x == 0 && tablet_y == 0) { // pen lifted off the tablet
```

Comment:
what is at tablet_x = 0 tablet_y = 0 there is a non-zero value, and the pen is actually there?

Reply:
a) it's probably not going to happen
b) it's on the edge. moving off the edge would send 0,0 anyway, so we're really just moving the edge a fraction of a millimeter if you want to think about it that way